lib/utils/ruleMessages.js

```diff
@@ -7,22 +7,25 @@
  * that has been violated.
  *
  * @param {string} ruleName
- * @param {object} messages - Object whose keys are message identifiers
+ * @param {{[k: string]: string|Function}} messages - Object whose keys are message identifiers
  *   and values are either message strings or functions that return message strings
- * @return {object} New message object, whose messages will be marked with the rule name
+ * @return {{[k: string]: string|Function}} New message object, whose messages will be marked with the rule name
  */
 module.exports = function(ruleName /*: string*/, messages /*: Object*/) /*: Object*/ {
-	return Object.keys(messages).reduce((newMessages, messageId) => {
-		const messageText = messages[messageId];
+	return Object.keys(messages).reduce(
+		(newMessages, messageId) => {
+			const messageText = messages[messageId];
 
-		if (typeof messageText === 'string') {
-			newMessages[messageId] = `${messageText} (${ruleName})`;
-		} else {
-			newMessages[messageId] = function() {
-				return `${messageText.apply(null, arguments)} (${ruleName})`;
-			};
-		}
+			if (typeof messageText === 'string') {
+				newMessages[messageId] = `${messageText} (${ruleName})`;
+			} else {
+				newMessages[messageId] = function() {
+					return `${messageText.apply(null, arguments)} (${ruleName})`;
+				};
+			}
 
-		return newMessages;
-	}, {});
+			return newMessages;
+		},
+		/** @type {{[k: string]: string|Function}} */ ({}),
```

Comment:
Why this type is needed here? It's an empty object, which is passed to `reduce` callback. Maybe callback function should have type for its parameters?

Reply:
take a look https://www.typescriptlang.org/play/index.html?target=99#code/GYVwdgxgLglg9mABAQwE6uQTwPLAHIgC2ARgKaoDOAFHMQFYBciA3gNoDWTFUqMYA5gF0myMJgC+ASiZgiZVK0EtEAWABQiTYlSkoIVEmz1S0AHTtSmarTqTTOgCYgIpKlWQQIAGkQXMkxABeAD4WdS0IlE9TAAcQCgALKnDI1Ji0ClIASTAoGnoOS0EfAEYABkkU1MkAbnUqyJ09AyiIOo0tcR9FWvVxerVQSFgEFHQsXAIScgoAYTh0EzybJjZORG5eAWEUMSkZOXJFZQam-UNjMz9rejtHZ1d3Tx8-AJCwjtSPCFj4pIbUpp0pRsrl8nRCphiohypVPpFemoAVozi1vu0Il1EMdkBRELJpgpBIj+ki1BAENxEDYgspkEwSj5iEwAEw+CBMADkAHpOYhxO00BgcPhDpRwYihRNRYS5gsdNAJTUgA

Reply:
What about this (`acc: number[]`)?

```ts
function arrayOfNumbersCorrect(obj: {[k: string]: any}): number[] { 
    return Object.keys(obj).reduce((acc: number[], key) => {
        acc.push(
            parseInt(obj[key], 10)
        );

        return acc;
    }, []);
}
```

https://www.typescriptlang.org/play/index.html?target=99&ssl=19&ssc=2&pln=11&pc=1#code/GYVwdgxgLglg9mABAQwE6uQTwPLAHIgC2ARgKaoDOAFHMQFYBciA3gNoDWTFUqMYA5gF0myMJgC+ASiZgiZVK0EtEAWABQiTYlSkoIVEmz1S0AHTtSmarTqTTOgCYgIpKlWQQIAGkQXMkxABeAD4WdS0IlE9TAAcQCgALKnDI1Ji0ClIASTAoGnoOS0EfAEYABkkU1MkAbnUqyJ09AyiIOo0tcR9FWvVxerVQSFgEFHQsXAIScgoAYTh0EzybJjZORG5eAWEUMSkZOXJFZQam-UNjMz9rejtHZ1d3TwPphWLfSwCQsI7UjwhYvEkg1Upp0pRsrl8nRCph3uVKr9Ir01CCtGcWv92hEuogeu1+qi1BAENxEDYgspkEwSj5iEwAEw+CBMADkAHpWYhxO00BgcPhDpRoSi+RNBa85gsdNARTUgA

I think it makes more sense.

Reply:
let's do it in this way =) Actually it is all about code style